### PR TITLE
Adding select decorator

### PR DESCRIPTION
**Pull request by kenotron**

### CODE REVIEW

File: packages/satcheljs/lib/select.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Decorator for action functions. Selects a subset from the state tree for the action.
+ */
+export default function select<StateShape>(selector: SelectorFunction) {
+    return function decorator<T extends Function>(target: T): T {
+        let _this = this;
+
+        let returnValue: any = function() {
+            let state = {};
+            let reaction = new Reaction('something', null);
```
Comment (jarmit) [inline packages/satcheljs/lib/select.ts]: What is 'something'. Seems like we should put something more descriptive

File: packages/satcheljs/lib/select.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    // If this is not an observable, then just create a getter
+                    Object.defineProperty(state, key, {
+                        get: () => selector[key]()
+                    });
+                }
+            });
+
+            reaction.dispose();
+
+            [].push.call(arguments, state);
```
Comment (jarmit) [inline packages/satcheljs/lib/select.ts]: This seems like weird syntax. Is there better way to do this

File: packages/satcheljs/lib/select.ts (hunk 1 above)
Comment (kenotron) [inline packages/satcheljs/lib/select.ts]: Oh haha! Yes, we should - but this is for internal tracking... supposed to just be unique. I'll add "Something" unique.

File: packages/satcheljs/lib/select.ts (hunk 2 above)
Comment (kenotron) [inline packages/satcheljs/lib/select.ts]: This is a pretty clean way to push an item to a collection within JS that isn't native array.

**Pull request merged**